src/main/java/betterquesting/client/gui2/editors/nbt/GuiQuestFrameSelection.java

```diff
@@ -0,0 +1,195 @@
+import betterquesting.api2.client.gui.resources.textures.ItemTexture;
+import betterquesting.api2.client.gui.themes.presets.PresetColor;
+import betterquesting.api2.client.gui.themes.presets.PresetTexture;
+import betterquesting.api2.utils.QuestTranslation;
+import net.minecraft.client.gui.GuiScreen;
+
+@SuppressWarnings("WeakerAccess")
+public class GuiQuestFrameSelection extends GuiScreenCanvas implements IPEventListener, IVolatileScreen {
+
+    private final ICallback<EnumFrameType> callback;
+    private final BigItemStack itemStack;
+    private final boolean isMain;
+    private final PanelButtonStorage<?>[] questStateButtons = new PanelButtonStorage<?>[EnumQuestState.values().length];
+    private EnumFrameType frameType;
+    private EnumQuestState questState = EnumQuestState.LOCKED;
+
+    private PanelQuest itemPreview;
+    private CanvasQuestFrame cvQuestFrame;
+
+    public GuiQuestFrameSelection(GuiScreen parent, EnumFrameType frameType, BigItemStack itemStack, ICallback<EnumFrameType> callback, boolean isMain) {
+        super(parent);
+        this.frameType = frameType;
+        this.itemStack = itemStack;
+        this.callback = callback;
+        this.isMain = isMain;
+    }
+
+    public void initPanel() {
+        super.initPanel();
+
+        PEventBroadcaster.INSTANCE.register(this, PEventButton.class);
+        Keyboard.enableRepeatEvents(true);
+
+        // Background panel
+        CanvasTextured cvBackground = new CanvasTextured(new GuiTransform(GuiAlign.FULL_BOX, new GuiPadding(0, 0, 0, 0), 0),
+                                                         PresetTexture.PANEL_MAIN.getTexture());
+        this.addPanel(cvBackground);
+
+        cvBackground.addPanel(new PanelButton(new GuiTransform(GuiAlign.BOTTOM_CENTER, -100, -16, 200, 16, 0), 0, QuestTranslation.translate("gui.done")));
+
+        PanelTextBox txTitle = new PanelTextBox(new GuiTransform(GuiAlign.TOP_EDGE, new GuiPadding(0, 16, 0, -32), 0),
+                                                QuestTranslation.translate("betterquesting.title.select_item")).setAlignment(1);
+        txTitle.setColor(PresetColor.TEXT_HEADER.getColor());
+        cvBackground.addPanel(txTitle);
+
+        // === LEFT PANEL ===
+
+        CanvasEmpty cvLeft = new CanvasEmpty(new GuiTransform(new Vector4f(0F, 0F, 0.2F, 1F), new GuiPadding(16, 32, 8, 8), 0));
+        cvBackground.addPanel(cvLeft);
+
+        PanelTextBox txSelection = new PanelTextBox(new GuiTransform(GuiAlign.TOP_EDGE, new GuiPadding(0, 0, 0, -16), 0),
+                                                    QuestTranslation.translate("betterquesting.gui.selection"));
+        txSelection.setColor(PresetColor.TEXT_MAIN.getColor());
+        cvLeft.addPanel(txSelection);
+
+        itemPreview = new PanelQuest(new GuiTransform(GuiAlign.TOP_LEFT, 0, 16, 36, 36, 0), 99, itemStack, frameType, questState, isMain);
+        cvLeft.addPanel(itemPreview);
+
+        EnumQuestState[] values = EnumQuestState.values();
```

Comment:
should include a note about what this is for (previewing state in different colors), feels a bit weird without any explanation